scene/3d/sprite_3d.cpp

```diff
@@ -522,7 +538,20 @@ void Sprite3D::_draw() {
 		}
 	}
 	set_aabb(aabb);
-	VS::get_singleton()->immediate_end(immediate);
+
+	Array mesh_array;
+	mesh_array.resize(VS::ARRAY_MAX);
+	mesh_array[VS::ARRAY_VERTEX] = mesh_vertices;
+	mesh_array[VS::ARRAY_NORMAL] = mesh_normals;
+	mesh_array[VS::ARRAY_TANGENT] = mesh_tangents;
+	mesh_array[VS::ARRAY_COLOR] = mesh_colors;
+	mesh_array[VS::ARRAY_TEX_UV] = mesh_uvs;
+
+	VS::get_singleton()->mesh_add_surface_from_arrays(mesh, VS::PRIMITIVE_TRIANGLE_FAN, mesh_array);
```

Comment:
I think you have code to update the mesh geometry from arrays directly, I would probably not use this.

Reply:
Also I would actually use encoded arrays, not given visualserver the task to encode every time.

Reply:
Are you talking about ``mesh_surface_update_region``?

Reply:
Done! I update the compressed arrays directly now.